### ISSUE TITLE
Fail to load right after a load

### ISSUE THREAD

**Issue by flaviojs** [freeorion/freeorion#2158]
### Environment
* **FreeOrion Version**: v0.4.8 (3f22a75)
* **Operating System**: Linux Mint 18 Sarah
* **Graphic card used**: Advanced Micro Devices, Inc. [AMD/ATI] RV710/M92 [Mobility Radeon HD 4530/4570/545v]
* **Fetched as**
  * Compiled from source

### Description
If I try to load another save right after a save is loaded, it will fail with:
![reload](https://user-images.githubusercontent.com/1009600/41496532-da75b2cc-7139-11e8-9563-e9d68ec4a7ab.png)
My guess is that the AIs are still thinking and the load procedure doesn't cancel them nor wait for them to finish.

### Expected Result
The load succeeds.

### Steps to reproduce
1. Start a new game and play a few turns.
2. Load an auto-saved turn.
3. Load another turn as quickly as possible.
4. Issue occurs.

**Comment by geoffthemedio** [comment]
Can't reproduce on windows with master.

Can you attach the log files from immediately after having this error appear?

**Comment by Vezzra** [comment]
What's the status of this? Any progress, can it be reproduced?

@flaviojs, can you reproduce the issue with the latest test builds?

**Comment by flaviojs** [comment]
Wow, I haven't been here in a long time. I'll try again.

**Comment by flaviojs** [comment]
Yup, I can still reproduce with b3daed1a47210ae1555e483e1092a2db08274f45.

Since this is a timing issue during load, to reproduce in fast computers you probably need to add a sleep (or a big loop) somewhere that doesn't affect the GUI to give you time to load another turn before the one you just loaded completely finishes whatever it is doing.

My laptop is on the low end spectrum as far as specs are concerned. (10 FPS)

**Comment by Vezzra** [comment]
Ok, thanks, so we'll keep that open.
